consensus/hotstuff/cruisectl/transition_time.go

```diff
@@ -17,6 +17,9 @@ var weekdays = map[string]time.Weekday{
 	strings.ToLower(time.Saturday.String()):  time.Saturday,
 }
 
+// epochLength is the length of an epoch (7 days, or 1 week).
+const epochLength = time.Hour * 24 * 7
```

Comment:
For now it's ok to keep it as constant but in future we can take this value from dynamic protocol state making it configurable.

Reply:
The end-goal is to take the target epoch switchover time from the protocol state directly, rather than doing this awkward inference. Then we won't need the notion of epoch length as a constant, we can just directly read the desired length in time of each epoch from the state.